cloud/amazon/lambda.py

```diff
@@ -0,0 +1,401 @@
+    code_b64 = base64.b64encode(code_hash)
+    hex_digest = code_b64.decode('utf-8')
+
+    return hex_digest
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        runtime=dict(type='str', required=True),
+        role_arn=dict(type='str', default=None),
+        handler=dict(type='str', default=None),
+        zip_file=dict(type='str', default=None, aliases=['src']),
+        s3_bucket=dict(type='str', default=None),
+        s3_key=dict(type='str', default=None),
+        s3_object_version=dict(type='str', default=None),
+        description=dict(type='str', default=''),
+        timeout=dict(type='int', default=3),
+        memory_size=dict(type='int', default=128),
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           mutually_exclusive=[['zip_file', 's3_key'],
+                                               ['zip_file', 's3_bucket'],
+                                               ['zip_file', 's3_object_version']])
+
+    name = module.params.get('name')
+    state = module.params.get('state').lower()
+    runtime = module.params.get('runtime')
+    role_arn = module.params.get('role_arn')
+    handler = module.params.get('handler')
+    s3_bucket = module.params.get('s3_bucket')
+    s3_key = module.params.get('s3_key')
+    s3_object_version = module.params.get('s3_object_version')
+    zip_file = module.params.get('zip_file')
+    description = module.params.get('description')
+    timeout = module.params.get('timeout')
+    memory_size = module.params.get('memory_size')
+
+    check_mode = module.check_mode
+    changed = False
+
+    if not HAS_BOTOCORE:
+        module.fail_json(msg='Python module "botocore" is missing, please install it')
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='Python module "boto3" is missing, please install it')
+
+    if not HAS_BINASCII:
+        module.fail_json(msg='Python module "binascii" is missing, please install it')
+
+    if not HAS_HASHLIB:
+        module.fail_json(msg='Python module "hashlib" is missing, please install it')
+
+    if not HAS_BASE64:
+        module.fail_json(msg='Python module "base64" is missing, please install it')
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg='region must be specified')
+
+    try:
+        client = boto3_conn(module, conn_type='client', resource='lambda',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
+        module.fail_json(msg=str(e))
+
+    current_function = get_current_function(client, name)
+    if not current_function:
```

Comment:
This conditional makes all runs fail - if I do:

```
  - lambda:
      state: present
      runtime: nodejs4.3
      handler: csv_read.handlers
      role_arn: arn:aws:iam::REDACTED:role/lambda-default-role
      name: myfunc
      profile: ansible-test
      zip_file: /home/ryansb/code/test-lambda/code.zip
```

For both `present` and `absent` state, I get the error `"msg": "Unable to get function information prior updating"`